CYBERRU-322 split DotaMatchSummary into multiple structures

kurenkoff opened the pull request.

Commit:
CYBERRU-322 split DotaMatchSummary into multiple structures

patstrom commented on structs/dota_match_summary.go:
```diff
@@ -0,0 +1,123 @@
+	Hero     Hero   `json:"hero"`
+}
+
+// RoshanEvent is a summarization of data about Roshan event of Dota match.
```
This seems to be the documentation for the RoshanEvent struct

kurenkoff replied:
Ye, funny copy-paste error

patstrom commented on structs/dota_match_summary.go:
```diff
@@ -0,0 +1,123 @@
+type DotaStructureDest struct {
+	Killer        int64  `json:"killer"`
+	StructureType string `json:"structure_type"`
+	StricturePos  string `json:"structure_pos"`
```
`StricturePos` should probably be `StructurePos`.

patstrom reviewed (changes requested):
Thank you for your PR! And thank you for fixing all the typos of "structure" that was in the old version. Something must've gone terribly wrong when auto-generating those.

Just a few typos and copy-paste errors to fix and I'm happy about this.

kurenkoff commented:
@patstrom Hey, thank you for your feedback. I've fixed errors and also added some fields which were declared in API Reference but not presented in go code (networth, camps_staked, wards, and runes)

patstrom reviewed (approved):
Thank you for your contribution!
